### ISSUE TITLE
Need to get the path to the `dotnet.exe` host in GetRunInformation MSBuild target

### ISSUE THREAD

**Issue by eerhardt** [dotnet/sdk#20]
We are currently returning `dotnet.exe` without a path in `<Target Name="GenerateRunInformation">` in the Microsoft.DotNet.Core.Sdk.targets file.

We should either:
1. Figure out a way to get to the correct `dotnet.exe` to call from inside our MSBuild target.
2. Or pass back some sort of indicator to `dotnet run` that this project should use its host to run the project.

**Comment by enricosada** [comment]
csc roslyn fixed embeeding a .cmd/.sh.
but other target files not bundled with sdk cannot do that, so ihmo is important.

i filled https://github.com/Microsoft/msbuild/issues/1669 in msbuild repo, because is not about sdk itself

**Comment by tannergooding** [comment]
I'm seeing an issue that is possible related:
1. Create a netcore console app
2. Explicitly target x86
3. Attempt to run
4. x64 dotnet is used and dotnet crashes

**Comment by livarcocc** [comment]
How are you invoking dotnet run? Is the version of the host that you are using a x86 host? Do you have the x86 host ahead on the PATH compared to the x64 one? dotnet run today simply shell out to the host. So, if you have a x86 project and you have an x64 host in the path (before the x86) you will hit this issue.

**Comment by tannergooding** [comment]
I just created a netcore console app, set it to target x86 and hit F5 (I have both the x64 and x86 SDK installed). This seems like a scenario that we should have covered, regardless of how the paths are configured.

**Comment by livarcocc** [comment]
This is something that we have been thinking about but we could not come up with the right solution yet. Strictly speaking, there is no way that we can know which dotnet in PATH is an x86 versus an x64 dotnet. So, this responsibility has been falling on the user.

I agree that if the user invoked <path_to_x86_dotnet>/dotnet run, then it should work, even if the x64 dotnet is ahead on the path, and I don't think that works today because of how we shell out the dotnet invocation. That we could fix if this bug were fixed.

**Comment by tannergooding** [comment]
Is this not something the SDK itself could expose, in a similar manner to how the SDK's are able to add new target frameworks (i.e. netcore2.0 shows in VS only after you install the 2.0 Preview SDK)? Otherwise, why not check the PE Header Metadata for all `dotnet` on the path... The first PE32+ version is used for x64 and the first PE32 version is used for x86...

**Comment by wli3** [comment]
Use linker's example. Use DOTNET_HOST_PATH on core. And use $(NetCoreRoot) with dotnet/dotnet.exe on fullframework. I believe $(NetCoreRoot) property is also available for Razor props and targets (please double check)

https://github.com/dotnet/sdk/blob/46917a9b622ea05377dc438db9f6b6e184bd3d83/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L78

**Comment by ViktorHofer** [comment]
Reopening as there's still a TODO in the sdk that points to this tracking issue: https://github.com/dotnet/sdk/blob/3ed908ee77c592bae787ac78f466c3bc0581970a/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L728

**Comment by eerhardt** [comment]
Can we at least remove this TODO?

https://github.com/dotnet/sdk/blob/ffaba7534225db5528a00d45be71127f3f871843/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets#L1421

**Comment by baronfel** [comment]
Yeah, we can flow through DOTNET_HOST_PATH here now that it's always set on both the framework and dotnet build pathways.
